src/Microsoft.AspNet.OData.Shared/Routing/Conventions/RefRoutingConvention.cs

```diff
@@ -69,7 +69,7 @@ internal static string SelectActionImpl(ODataPath odataPath, IWebApiControllerCo
                         controllerContext.AddKeyValueToRouteData((KeySegment)odataPath.Segments[1]);
                     }
 
-                    controllerContext.RouteData.Add(ODataRouteConstants.NavigationProperty, navigationLinkSegment.NavigationProperty.Name);
+                    controllerContext.AddNavigationPropertyToRouteData(navigationLinkSegment);
```

Comment:
>AddNavigationPropertyToRout [](start = 38, length = 27)

shall we modify NavigationRoutingConvention.cs

Reply:
What do you mean by this?

I added the `AddNavigationPropertyToRouteData()` extension method to encapsulate the adding the navigation property from the link segment to the route data manually. And also to ensure that the key count was incremented.